src/components/Todo/TodoItem.tsx

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import * as React from 'react';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+import { updateTodo, deleteTodo } from '../../api/todos';
+import { TodoItemProps } from '../../types/types';
+
+type UpdateEvent =
+  | React.MouseEvent<HTMLInputElement, MouseEvent>
+  | React.FormEvent<HTMLFormElement>;
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  todo,
+  todos,
+  setTodos,
+}) => {
+  const [loader, setLoader] = React.useState(false);
+  const [isFocused, setIsFocused] = React.useState(false);
+  const [title, setTitle] = React.useState(todo.title);
+
+  const newList = React.useMemo(() => {
+    return todos.filter(item => item.id !== todo.id);
+  }, [todos, todo.id]);
+
+  function handleUpdateTodo(newData: string | boolean, event: UpdateEvent) {
+    event.preventDefault();
+
+    const preparedTodos = (updatedTodo: Todo) => {
+      return todos.map(item => {
+        return item.id === updatedTodo.id ? updatedTodo : item;
+      });
+    };
+
+    setLoader(true);
+
+    switch (newData) {
+      case todo.completed:
+        updateTodo(todo.id, {
+          ...todo,
+          completed: !todo.completed ? true : false,
+        }).then(res => {
+          setTodos(preparedTodos(res));
+
+          setLoader(false);
+        });
+        break;
+      case todo.title:
+        updateTodo(todo.id, {
+          ...todo,
+          title: title,
+        }).then(res => {
+          setTodos(preparedTodos(res));
+
+          setLoader(false);
+          setIsFocused(false);
+        });
+        break;
+    }
+  }
+
+  function handleDeleteTodo(
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) {
+    event.preventDefault();
+
+    setLoader(true);
+
+    deleteTodo(todo.id)
+      .then(() => {
+        setTodos(newList);
+        setLoader(false);
+      })
+      .catch(() => {
+        setLoader(false);
+      });
+  }
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames('todo', {
+        completed: todo.completed,
+      })}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          defaultChecked={todo.completed}
```

Comment:
Consider using [destruction assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) for `todo` object